app_integrations/apps/salesforce.py

```diff
@@ -0,0 +1,456 @@
+            return True
+
+    def _list_log_files(self):
+        """Fetch a list of available log files by event types.
+
+        An event generates log data in real time. However, log files are generated
+        the day after an event takes place, during nonpeak hours. Therefore, log
+        file data is unavailable for at least one day after an event.
+
+        Returns:
+            list: a list of log file location or empty list.
+
+        An example of log files response json body:
+        {
+            "totalSize": 2,
+            "done": True,
+            "records": [
+                {
+                    "attributes": {
+                        "type": "EventLogFile",
+                        "url": "/services/data/v32.0/sobjects/EventLogFile/0ATD00001bROAQ"
+                    },
+                    "Id": "0ATD000000001bROAQ",
+                    "EventType": "Console",
+                    "LogFile": "/services/data/v32.0/sobjects/EventLogFile/0ATD00001bROAQ/LogFile",
+                    "LogDate": "2014-03-14T00:00:00.000+0000",
+                    "LogFileLength": 2692.0
+                },
+                {
+                    "attributes": {
+                        "type": "EventLogFile",
+                        "url": "/services/data/v32.0/sobjects/EventLogFile/0ATD000000001SdOAI"
+                    },
+                    "Id": "0ATD000000001SdOAI",
+                    "EventType": "Console",
+                    "LogFile": "/services/data/v32.0/sobjects/EventLogFile/0ATD00001SdOAI/LogFile",
+                    "LogDate": "2014-03-13T00:00:00.000+0000",
+                    "LogFileLength": 1345.0
+                }
+            ]
+        }
+        """
+        event_type_format = 'AND+EventType+=+\'{}\''.format(self._type)
+        url = self._SALESFORCE_QUERY_URL.format(
+            instance_url=self._instance_url,
+            api_version=self._latest_api_version,
+            query=self._SALESFORCE_QUERY_FILTERS,
+            start_time=self._SALESFORCE_CREATE_AFTER.format(self._last_timestamp),
+            event_type=event_type_format
+        )
+        success, response = self._make_get_request(url, self._auth_headers)
+        if not success:
+            LOGGER.error('Failed to get a list of log files.')
+            return
+
+        log_files = []
+        if response.get('records'):
+            log_files.extend([record['LogFile'] for record in response['records']
+                              if record.get('LogFile')])
+
+        LOGGER.debug('Retrived %d log files', len(log_files))
+        return log_files
+
+    def _fetch_event_logs(self, log_file_path):
+        """Get event logs by sending GET request to each log file location.
+
+        Args:
+            log_file_path (str): log file location.
+
+        Returns:
+            list: a list of event logs or None.
+        """
+        url = '{}/{}'.format(self._instance_url, log_file_path)
+        data = None
+        try:
```

Comment:
what do you think about changing this try.. I think the StringIO stuff might be unnecessary 🤔. isn't `resp` just a string?

you might be able to just do:
```
        try:
            success, resp = self._make_get_request(url, self._auth_headers)
        except SalesforceAppError:
            LOGGER.exception('An exception occurred when getting event logs')

        if not (success and resp):
            LOGGER.error('Failed to get event logs')
            return

        return resp.splitlines()[1:]
```

Reply:
You are right! Using StringIO is over kill.